soh/src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -296,6 +296,7 @@ void GivePlayerRandoRewardSheikSong(EnXc* sheik, GlobalContext* globalCtx, Rando
         GetItemID getItemId = Randomizer_GetItemIdFromKnownCheck(check, ogSongId);
         if (check == RC_SHEIK_AT_TEMPLE && !Flags_GetTreasure(globalCtx, 0x1F)) {
             if (func_8002F434(&sheik->actor, globalCtx, getItemId, 10000.0f, 100.0f)) {
+                player->getItemEntry = Randomizer_GetItemFromKnownCheck(check, ogSongId);
```

Comment:
we already have the `getItemId` from line 296 here, is there a reason to call into this again?

Reply:
That's not getting the getItemId, that's getting the getItemEntry. I'm not finished with the entire conversion yet, but eventually the idea is that we only need to call this and not the one for the getItemId. That won't change how people register items to the table though or how they apply says items to Link's inventory, which is why I wanted to get this PR out there, since we've got people making keysanity and other things which would benefit greatly from this.

Reply:
ah, missed that (names are very similar) so if we changed line 296 here to get the entry, then replaced getItemId in line 298 with the id from the entry, we could avoid the extra call

Reply:
Eventually yeah, but I think some things in z_player.c still depend on having player->getItemId so I'd need to change those first. I could probably add the getItemId to the GetItemEntry struct though as a way around it.

Reply:
yeah, forgot the entry didn't include the getItemId for a second, the itemid vs getitemid stuff strikes again! this is fine as is, but yeah, maybe adding the getitemid to the struct could be a cool future thing

Reply:
It'd be easy enough to do I might as well cut out the extra calls while I'm fixing the other things you mentioned. I'll be able to get on it soon now that custom messages is merged.

Reply:
Addressed in 11db45eef6d6c7cbd2b438de5a69665ce0544738